### PR TITLE
fix-list-objects-delimiter-none cut out None from Delimiter tag

### PR DESCRIPTION
**Pull request by yaitskov**
**Please refer to the contribution guidelines in the README when submitting PRs.**

### CODE REVIEW

File: localstack/services/s3/s3_listener.py [hunk 1]
```diff
@@ -500,6 +500,21 @@ def fix_creation_date(method, path, response):
         r'\1Z</CreationDate>', to_str(response._content))
 
 
+CUT_DELIMITER_NONE_RE_STR = re.compile('[<]Delimiter[>]None[<]')
```
[the remark this answers is not in the document]
Comment (yaitskov) [reply localstack/services/s3/s3_listener.py:503]: ok. I will rewrite, you version looks more dry but is it better in performance terms?

File: localstack/services/s3/s3_listener.py [hunk 2]
```diff
@@ -500,6 +500,21 @@ def fix_creation_date(method, path, response):
...
+CUT_DELIMITER_NONE_RE_BYTES = re.compile(b'[<]Delimiter[>]None[<]')
+
+def fix_delimiter(data, headers, response):
+    if response.status_code == 200:
+        c = response._content
+        if isinstance(c, bytes) and c.startswith(b'<?xml'):
```
[the remark this answers is not in the document]
Comment (yaitskov) [reply localstack/services/s3/s3_listener.py:509]: where did you get to_bytes functions? Regex has to be compiled. Your version doesn't do this. I have no clue how to  convert string regex into  byte regex. Compiling pattern on every request must be slow.

[the remark this answers is not in the document]
Comment (yaitskov) [reply localstack/services/s3/s3_listener.py]: I rewrote the patch according to your recommendation.
IMHO it  looks horrible.

**Comment by whummer** [comment]
Thanks for the quick turnaround @yaitskov . (Btw, Python performs caching of compiled regexes internally, so clarity should be prioritized over performance here.) Thanks

**Pull request merged**